src/addBusinessDays/index.ts

```diff
@@ -37,110 +36,63 @@ export default function addBusinessDays(
   requiredArgs(2, arguments)
   const options = dirtyOptions || {}
   const exceptions = options.exceptions || {}
-  const businessDays =
-    options.businessDays == null
-      ? [1, 2, 3, 4, 5]
-      : options.businessDays.filter((number) => number < 7).map(toInteger)
+  const businessDays = options.businessDays ?? [1, 2, 3, 4, 5]
 
-  const date = toDate(dirtyDate)
-  const amount = toInteger(dirtyAmount)
-  let accountedForExceptions = 0
-  const isExcepted = (date: Date): boolean | null => {
-    if (options.exceptions) {
-      const exception =
-        Object.keys(options.exceptions).find((e) =>
-          isSameDay(new Date(e), date)
-        ) || ''
-      if (options.exceptions[exception] !== undefined) {
-        accountedForExceptions += 1
-      }
-      return options.exceptions[exception]
-    }
-    return null
+  // Throw an exception if businessDays includes a number greater than 6
+  if (businessDays?.filter((number) => number > 6).length > 0) {
+    throw new RangeError('business days must be between 0 and 6')
   }
-  const isNonWorkingDay = (date: Date) =>
-    isExcepted(date) === false || !businessDays.includes(date.getDay())
+
+  const initialDate = toDate(dirtyDate)
+  const amount = toInteger(dirtyAmount)
 
   if (isNaN(amount)) return new Date(NaN)
 
-  if (date.toString() === 'Invalid Date') {
-    return date
+  if (initialDate.toString() === 'Invalid Date') {
+    return initialDate
   }
-  const startedOnNonWorkingDay = isNonWorkingDay(date)
-  const hours = date.getHours()
-  const sign = amount < 0 ? -1 : 1
-  const fullWeeks = toInteger(amount / businessDays.length)
-  date.setDate(date.getDate() + fullWeeks * 7)
 
-  // Get remaining days not part of a full week
-  let restDays = Math.abs(amount % businessDays.length)
-  // Loops over remaining days
-  while (restDays > 0) {
-    date.setDate(date.getDate() + sign)
-    if (!isNonWorkingDay(date)) restDays -= 1
+  // returns true or false if the date has an exception, else null
+  const findException = (date: Date): boolean | null => {
+    if (options.exceptions) {
+      const exceptionDate =
+        Object.keys(exceptions).find((e) => isSameDay(new Date(e), date)) || ''
+      return exceptions[exceptionDate]
+    }
+    return null
   }
```

Comment:
request change: This feels like an inefficient way to find the exception for a given day (I would expect, at least if the number of exceptions is large). The exceptions are keyed by date, so it would probably be worth converting the provided date to the format used by the exceptions object as keys and seeing if it's there without the find.

Also, returning `boolean | undefined` might be easier, since undefined is what would be returned should the exception not exist.